packages/ui/src/ValueViewComponent/sampleValueViews.ts

```diff
@@ -0,0 +1,111 @@
+import {
+  Metadata,
+  ValueView,
+} from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/asset/v1/asset_pb';
+import { bech32mIdentityKey } from '@penumbra-zone/bech32m/penumbravalid';
+
+const u8 = (length: number) => Uint8Array.from({ length }, () => Math.floor(Math.random() * 256));
+const validatorIk = { ik: u8(32) };
+const validatorIkString = bech32mIdentityKey(validatorIk);
+const delString = 'delegation_' + validatorIkString;
+const udelString = 'udelegation_' + validatorIkString;
+const delAsset = { inner: u8(32) };
+const unbondString = 'unbonding_start_at_123_' + validatorIkString;
+const uunbondString = 'uunbonding_start_at_123_' + validatorIkString;
+const unbondAsset = { inner: u8(32) };
+
+const DELEGATION_TOKEN_METADATA = new Metadata({
+  display: delString,
+  base: udelString,
+  denomUnits: [{ denom: udelString }, { denom: delString, exponent: 6 }],
+  name: 'Delegation token',
+  penumbraAssetId: delAsset,
+  symbol: 'delUM(abc...xyz)',
```

Comment:
issue: the symbol should be truncated

<img width="1064" alt="image" src="https://github.com/user-attachments/assets/609a2ee1-fe3a-4292-8d2e-9f457d657868">

previously, we removed any programmatic truncation from the token symbols, and they theoretically can be long enough to not fit into the view. This way, CSS truncation is needed.

and from a11y point, when truncation happens, the `title` attribute is the best to show the full symbol in the tooltip

Reply:
Hm... [we’re still programmatically truncating _symbols_](https://github.com/penumbra-zone/web/blob/cbc2419/packages/wasm/crate/src/metadata.rs#L171), although perhaps not _display names_, of assets. And it’s not just at runtime — the programmatically truncated symbols are stored in the database that way as well.

We could still put a max-width + text-overflow on these, but that has security implications: if two symbols start with the same text, and we don't show the complete text, users could think they're e.g., sending one token when they're actually sending a different one. If we show the complete symbol, that can't happen. (Granted, we're truncating the symbol when it includes e.g., a validator ID, but those shouldn't have any collisions.)

I'm tempted to leave it as-is, to avoid this security issue. Then, consumers of the component would be forced to accommodate a variable width in their layout (possibly by adding a horizontal scrollbar if needed).

Your thoughts?

Reply:
Interestingly, the tests in Wasm package fail. They are not integrated into the CI, so we didn't even know about it (created issue #1546). I forgot to change them in [this commit](https://github.com/penumbra-zone/web/commit/cbc241931c45651ffbc31d5bd4f8db2dec4f0164) that removed the truncation. 

Agree on the risk of showing the equal first park of two symbols. Maybe we could programmatically truncate the inner part of the braces on the client and, for example, show it on hover?

```html
<p>
  <span>unboundUMat123(abc</span>
  <span title="unboundUMat123(abcdefghklmnxyz)">...</span>  <!-- Notice the title that could be a tooltip  -->
  <span>xyz)</span>  
</p>
```

Reply:
Oh! Didn't realize the implementation had changed.

OK, hm. Since we don't know that all symbols will have braces like the unbonding/delegation tokens, I'd rather not programmatically truncate them. Instead, I've just pushed up [a commit](https://github.com/penumbra-zone/web/pull/1526/commits/35ad53e71cf5277451b8c72dd3ddd32fc87a8716) that truncates it with CSS and then adds a `title` with the full symbol. 

There is [this weird way](https://stackoverflow.com/a/10017343/974981) to detect if the CSS ellipsis has been applied. I'd love to do that, and show a tooltip in that case, but I'd prefer to do that in a separate PR.

(Re: the build failures — there appears to be a CI issue. Resolving that separately with L...)

Reply:
Screenshot:

<img width="489" alt="image" src="https://github.com/user-attachments/assets/ec1a47bf-a29b-4152-acf9-1bc6b0873698">